docs/integrations/data-ingestion/clickpipes/mongodb/source/generic.md

```diff
@@ -0,0 +1,57 @@
+---
+sidebar_label: 'Generic MongoDB'
+description: 'Set up any MongoDB instance as a source for ClickPipes'
+slug: /integrations/clickpipes/mongodb/source/generic
+title: 'Generic MongoDB source setup guide'
+---
+
+# Generic MongoDB source setup guide
+
+:::info
+
+If you use MongoDB Atlas, please refer to the specific guide [here](./atlas).
+
+:::
+
+## Enable oplog retention {#enable-oplog-retention}
+
+Minimum oplog retention of 24 hours is required for replication. We recommend setting the oplog retention to 72 hours or longer to ensure that the oplog is not truncated before the initial snapshot is completed.
+
+You can check your current oplog retention by running the following command in the MongoDB shell (you must have `clusterMonitor` role to run this command):
+
+```javascript
+db.serverStatus().oplogTruncation.oplogMinRetentionHours
+```
+
+To set the oplog retention to 72 hours, run the following command as an admin user:
+
+```javascript
+db.adminCommand({
+    "replSetResizeOplog" : 1,
```

Comment:
> replSetResizeOplog Does Not Replicate To Other Members
Changing the oplog size or minimum oplog retention period of a given replica set member with replSetResizeOplog does not change the oplog size of any other member in the replica set. You must run replSetResizeOplog on each replica set member in your cluster to change the oplog size or minimum retention period for all members.

https://www.mongodb.com/docs/manual/reference/command/replSetResizeOplog/#replsetresizeoplog-does-not-replicate-to-other-members

Maybe direct them to config or see if there's a way to run a command on all replicas?

Reply:
great catch.